cli/azd/cmd/util.go

```diff
@@ -217,6 +218,11 @@ func ensureEnvironmentInitialized(
 	hasSubID := hasValue(environment.SubscriptionIdEnvVarName)
 	hasPrincipalID := hasValue(environment.PrincipalIdEnvVarName)
 
+	if tenantId, found := env.Config.Get(azdEnvTenant); found {
+		// ask azCli to use tokens for this tenant id for the rest of this command
+		azCli.SetTenantId(tenantId.(string))
```

Comment:
I haven't dug deep enough, but overall feeling is that we're exposing too much flexibility everywhere.

Most code shouldn't need to know about how to correctly obtain a user's credential. Would it be possible to invert:

1. Somehow, `login` + `init` persists both multi-tenant and single-tenant credential. The single-tenant credential is flown and injected by default.
2. The subscription listing should ignore single-tenant and only use a multi-tenant credential.
3. The rest of cli should not need to know about which credential to use. The cred provider is available if needed, but by default this should just consume the default credentials as-is.